gobblin-config-management/gobblin-config-core/src/main/java/gobblin/config/common/impl/SingleLinkedListConfigKeyPath.java

```diff
@@ -0,0 +1,52 @@
+package gobblin.config.common.impl;
+
+import gobblin.config.store.api.ConfigKeyPath;
+
+public class SingleLinkedListConfigKeyPath implements ConfigKeyPath{
+  
+  public static final SingleLinkedListConfigKeyPath ROOT = new SingleLinkedListConfigKeyPath(null, "");
+
+  private final ConfigKeyPath parent;
+  private final String ownName;
+  
+  // constructor private, can only create path from ROOT using createChild method
+  private SingleLinkedListConfigKeyPath(ConfigKeyPath parent, String name){
+    this.parent = parent;
+    this.ownName = name;
+  }
+  
+  @Override
+  public ConfigKeyPath getParent() {
+    return this.parent;
+  }
+
+  @Override
+  public String getOwnPathName() {
+    return this.ownName;
+  }
+
+  @Override
+  public ConfigKeyPath createChild(String childPathName) {
+    return new SingleLinkedListConfigKeyPath(this, childPathName);
+  }
+
+  @Override
+  public String getAbsolutePathString() {
+    if(this.isRootPath()){
+      return "";
```

Comment:
This is trivial to avoid. Regardless of implementation details, the right semantics is to return "/".